test/e2e/denylist_test.go

```diff
@@ -38,7 +38,7 @@ func testDenyPrometheus(t *testing.T) {
 
 	ctx.SetupPrometheusRBAC(t, operatorNamespace, framework.KubeClient)
 
-	_, err := framework.CreatePrometheusOperator(operatorNamespace, *opImage, nil, deniedNamespaces, nil, nil, false)
```

Comment:
While testing I noticed that deleting a ServiceMonitor does not actually update the targets correctly. I would suggest we add a test for this, seems like a regression as before it reconciled that correctly.

This might be the case for other resources, did not test, but might we work looking at other CRs.

Reply:
Interesting, let me add a test case for this.

Reply:
Works for me when testing manually, but adding a test is definitely a good idea. cc'ing @simonpasquier as i believe lili saw the latency by configmap-reloader which is currently being refactored.

Reply:
Added tests.